src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -17,202 +17,191 @@
+                                             // Handle case where key is not found
+                                             value = F("-256"); // Placeholder value
                                            } else {
-                                             startIndex += key.length() + 2; // Move index past the key
-                                             endIndex    = str.indexOf(',', startIndex);
+                                             int endIndex = 0;
+
+                                             if (!equals(eventName, F("current"))) {
+                                               // In daily and hourly, the values are stored in an array
+                                               startIndex += key.length() + 3; // Move index past the key
+                                               endIndex    = str.indexOf(']', startIndex);
+                                             } else {
+                                               startIndex += key.length() + 2; // Move index past the key
+                                               endIndex    = str.indexOf(',', startIndex);
+                                             }
+
+                                             // Find the index of the next comma
+                                             if ((endIndex == -1) || (endIndex > str.indexOf("}", startIndex))) {
+                                               endIndex = str.indexOf('}', startIndex); // If no comma is found or comma comes after },
+                                                                                        // take the rest of the string
+                                             }
+
+                                             value = str.substring(startIndex, endIndex);
+                                             value.trim(); // Remove any surrounding whitespace
                                            }
 
-                                           // Find the index of the next comma
-                                           if ((endIndex == -1) || (endIndex > str.indexOf("}", startIndex))) {
-                                             endIndex = str.indexOf('}', startIndex); // If no comma is found or comma comes after },
-                                                                                      // take the rest of the string
+                                           if (!csv.isEmpty()) {
+                                             csv += ',';
                                            }
-
-                                           value = str.substring(startIndex, endIndex);
-                                           value.trim(); // Remove any surrounding whitespace
-                                         }
-
-                                         if (!csv.isEmpty()) {
-                                           csv += ',';
+                                           csv += value;
```

Comment:
Shouldn't this also check for presence of any space or comma?
There is a function to wrap a string in quotes (not used in that string) whenever a possible separator is present.

Have to look for the actual function name...

Reply:
Ah found it...
`wrapWithQuotesIfContainsParameterSeparatorChar`